### PR TITLE
Spring Web DTO

### PR DESCRIPTION
**Pull request by SergeyTsimbal**

### CODE REVIEW

File: src/main/java/com/example/springbootbookstore/service/BookServiceImpl.java [hunk 1]
```diff
@@ -1,28 +1,40 @@
...
+        return bookMapper.toDto(bookRepository.save(book));
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
```
Comment (MAMentorFX) [inline src/main/java/com/example/springbootbookstore/service/BookServiceImpl.java]: Mapstruct supports mapping collections from the box

**Review by MAMentorFX — CHANGES_REQUESTED** [review]
Almost excelent :)

File: src/main/java/com/example/springbootbookstore/service/BookServiceImpl.java (hunk 1 above)
Comment (SergeyTsimbal) [reply src/main/java/com/example/springbootbookstore/service/BookServiceImpl.java]: I added List to mapper

File: src/main/java/com/example/springbootbookstore/controller/BookController.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/books")
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/springbootbookstore/controller/BookController.java]:
```suggestion
@RequestMapping("/books")
```

File: src/main/java/com/example/springbootbookstore/controller/BookController.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+public class BookController {
+
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/springbootbookstore/controller/BookController.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/com/example/springbootbookstore/controller/BookController.java (hunk 3 above)
Comment (SergeyTsimbal) [reply src/main/java/com/example/springbootbookstore/controller/BookController.java]: Сorrected

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**